src/ADR62.ts

```diff
@@ -0,0 +1,16 @@
+import { keccak256 } from "ethereum-cryptography/keccak"
+import { toHex } from "ethereum-cryptography/utils"
+
+/**
+ * Calculates the metadata hash. Uses the keys to determine which fields of the metadata object will be used for the result.
+ * @public
+ */
+export function keccak256Hash(metadata: any, keys: string[]): string {
+  const partialMetadata = JSON.stringify(pick(metadata, keys.sort()))
```

Comment:
We must be aware that the keys are being sort here. If we leave this as is (and it's a good idea to do it), we'll need to write that in the ADR62.